lib/controller.js

```diff
@@ -641,6 +663,213 @@ controller.tesselFirmwareVerion = function(opts) {
+      if (label[0] !== 'E') {
+        callback(tessels[index].name, index, resolve, reject);
+        // controller.ssh.runSSH(index, opts, tessels, resolve, reject);
+      } else {
+        // going to clear screen and calling exit to resolve promise
+        controller.menu.clear();
+        controller.menu.exit(resolve);
+      }
+    });
+    //  });
+  },
+  makeMenu: function(tessel, index, rtm) {
+    if (tessel.lanConnection) {
+      if (!tessel.lanConnection.authorized) {
+        rtm.add(index + ') ' + tessel.name + ': ' + tessel.lanConnection.ip + ' (not authorized) \n');
+      } else {
+        rtm.add(index + ') ' + tessel.name + ': ' + tessel.lanConnection.ip + ' (authorized) \n');
+      }
+
+    } else if (tessel.usbConnection) {
+      if (!tessel.usbConnection.authorized) {
+        rtm.add(index + ') ' + tessel.name + ': [USB] (not authorized) \n');
+      } else {
+        rtm.add(index + ') ' + tessel.name + ': [USB] (authorized) \n');
+      }
+    }
+    // FIXME: Workaround for trouble with dynamic added menu elements
+    rtm._draw();
+  },
+  showMenu: function(rtm, resolve, reject) {
+    // Menu navigation
+    process.stdin.pipe(rtm.createStream()).pipe(process.stdout);
+
+    // raw mode for gain ability to navigate up and down within the menu
+    process.stdin.setRawMode(true);
+
+    // cb 
+    rtm.on('close', function() {
+      process.stdin.setRawMode(false);
+      process.stdin.end();
+    });
+    rtm.on('error', function(e) {
+      process.stdin.setRawMode(false);
+      reject(e);
+    });
+  },
+  seek: function(opts) {
+    // to be able to replace the seeker in testing mode
+    // return Tessel.seekTessels(opts);
+    return Tessel.get(opts);
+  },
+  clear: function() {
+    // selected exit!
+    process.stdout.write('\u001b[2J\u001b[0;0H');
+  },
+  exit: function(resolve) {
+    resolve();
+    //process.exit();
+  }
+};
+controller.ssh = {
+  runSSH: function(id, opts, tessels, resolve, reject) {
+    // clear console
+    debug('runSSH...\n root@' + tessels[id].lanConnection.host);
+    if (opts.menu) {
+      // because the test needs to resolve the promise ...
+      resolve({
+        opts: opts,
+        tessels: tessels
+      });
+    } else {
+      // if this is no test, starting the menu as child process
+      controller.menu.clear();
+      var ch = require('child_process')
+        .spawn('ssh', ['-i',
+          opts.path,
+          'root@' + tessels[id].lanConnection.ip
+        ], {
+          stdio: 'inherit'
+        });
+      logs.info('Connect to ' + tessels[id].lanConnection.host + '...');
+
+      // FIXME: There is no handler for poweroff Tessel ! Needs to be fixed within the firmware...
+      // (Terminal freezes imediatelly after poweroff the Tessel)
+
+      ch.once('error', function(e) {
+        if (e === 255) {
+          controller.menu.clear();
+          controller.ssh.notAuthorized();
```

Comment:
the `t2 root` command isn't interested in whether a Tessel 2 is authorized. It would be able to use a different `id_rsa` by `t2 root -i http://local/key_from_cloud` (e.g.) and login to a Tessel. In the case the key is wrong ... this error code is responded. 

At this point: Once more ... the solution with a flag "authorized" isn't helpful for creating a menu or do other things like replacing id_rsa ...

I think at the moment I need to create
```
controller.ssh = {
// ....
  checkAuthorized = function(ip,id_rsa){
   var ch = require('child_process')
        .spawn('ssh', ['-i',
         id_rsa,
          'root@' + ip
        ], {
          stdio: 'inherit'
        });
    ch.once('error', function(e) {
        if (e === 255) {
          return false;
       }
      });
    ch.once('exit', function(e) {
     if (e === 255) {
      return false;   
      } else if(e === 0){
      return true;
    }
 });
ch.kill('SIGHUP');
  }
}

```
And call it within 
```
controller.menu = {
// ...
 makeMenu: function(tessel, index, rtm) {
    if (tessel.lanConnection) {
var auth = controller.ssh.checkAuthorized(tessel.lanConnection.ip,opts.path)
// if (!tessel.lanConnection.authorized) {
if (!auth) {
rtm.add(index + ') ' + tessel.name + ': ' + tessel.lanConnection.ip + ' (not authorized) \n');
      } else {
        rtm.add(index + ') ' + tessel.name + ': ' + tessel.lanConnection.ip + ' (authorized) \n');
      }
// ...
```